types/prosemirror-view/index.d.ts

```diff
@@ -568,7 +568,7 @@ export interface EditorProps<ThisT = unknown, S extends Schema = any> {
    */
   nodeViews?: {
     [name: string]: (
-      node: ProsemirrorNode<S>,
+      node: ProsemirrorNode<S> | Mark,
```

Comment:
Perhaps make this generic.

```suggestion
      nodeOrMark: ProsemirrorNode<S> | Mark<S>,
```

Reply:
Making the `mark` type generic is good, though the `nodeOrMark` name is a bit clunky. I think a better solution would be to separate the two entirely by creating a type definition for `MarkView`, so

```ts
  nodeViews?: {
    [name: string]: (
    (
      node: ProsemirrorNode<S>,
      view: EditorView<S>,
      getPos: (() => number),
      decorations: Decoration[]
    ) => NodeView<S>) |
   (
      mark: Mark<S>,
      view: EditorView<S>,
      getPos: boolean,
      decorations: Decoration[]
   ) => MarkView<S>
 )
  } | null;
```

This is more correct, since there will never be a custom NodeView with getPos being a boolean, and a custom Markview with getPos being a function.

Reply:
I think union declarations of function properties can cause problems. It's been a while since I tried to use one so it might no longer be an issue in newer versions of TypeScript. 

Last time I checked the way that TypeScript merges the methods is to create an intersection of each positional argument. The type declared would become:

```ts
{
  nodeViews?: {
    [key: string]: (
      arg: ProsemirrorNode & Mark,
      view: EditorView,
      getPos: ((() => number) & false) | ((() => number) & true),
      decorations: Decoration[],
    ) => NodeView<S> & MarkView<S>;
  };
}
```

If this is the case, the automatic intersection types would make it unusable.

I really like the idea of adding a `MarkView` type.

Reply:
I added a `MarkView` type which seems to work. However, testing that types worked correctly required Typescript 3.8.